plugin-infra/go-plugin-access/src/com/thoughtworks/go/plugin/access/authorization/AuthorizationPluginInfoBuilder.java

```diff
@@ -41,10 +41,9 @@ public AuthorizationPluginInfo pluginInfoFor(GoPluginDescriptor descriptor) {
 
         PluggableInstanceSettings authConfigSettings = authConfigSettings(descriptor.id());
         PluggableInstanceSettings roleSettings = roleSettings(descriptor.id(), capabilities);
-        PluggableInstanceSettings pluginSettingsAndView = getPluginSettingsAndView(descriptor, extension);
         Image image = image(descriptor.id());
 
-        return new AuthorizationPluginInfo(descriptor, authConfigSettings, roleSettings, image, capabilities, pluginSettingsAndView);
+        return new AuthorizationPluginInfo(descriptor, authConfigSettings, roleSettings, image, capabilities, null);
```

Comment:
This and [this](https://github.com/gocd/gocd/pull/3865/files#diff-4604837e13357ba5ea488420ebdbac1eR37) are not correct. The auth plugins written/used by us so far do not allow plugin-settings, but that doesn't mean that the authorization-extension itself doesn't support plugin-settings.

Reply:
Even if a plugin-extension supports settings, a plugin could choose not to implement it, so the exception stacktrace from the server startup won't really go away with this. One option could be to mandate plugins to provide a list of capabilities - settings being one of them. We could invoke calls only if the plugin reports having a capability to configure settings. Another option, given the lack of the first one is to use [FelixGoPluginOSGiFramework.doOnWithExceptionHandling](https://github.com/gocd/gocd/blob/master/plugin-infra/go-plugin-infra/src/com/thoughtworks/go/plugin/infra/FelixGoPluginOSGiFramework.java#L275) for plugin-settings call, the [handler](https://github.com/gocd/gocd/blob/master/plugin-infra/go-plugin-infra/src/com/thoughtworks/go/plugin/infra/FelixGoPluginOSGiFramework.java#L307) could return null or ignore a plugin if the call is not implemented by it.